### PR TITLE
DEVPROD-4767: support tenancy option for host.create

### PR DESCRIPTION
**Pull request by Kimchelly**
DEVPROD-4767

### Description
Allow users to specify the EC2 host [tenancy option](https://docs.aws.amazon.com/autoscaling/ec2/userguide/auto-scaling-dedicated-instances.html) for host.create. Will add a follow-up PR in Shrub to add the new option once this is merged.

### Testing
* Added unit tests.
* Verified in staging that [a task could start an EC2 host.create host with tenancy set](https://spruce-staging.corp.mongodb.com/task/evg_ubuntu2204_kim_task_patch_3a552c8209cca308b80b52627897a8cec3000e2b_65ce55f7b8f8990007c8930c_24_02_15_18_20_39/logs?execution=1). I also double-checked that `aws ec2 describe-instances` for the instance shows its tenancy is `dedicated` (the custom value I set in the task) instead of `shared` (the default).

### Documentation
Updated docs.

### CODE REVIEW

File: apimodels/agent_models.go [hunk 1]
```diff
@@ -243,6 +246,9 @@ func (ch *CreateHost) validateEC2() error {
 			catcher.New("subnet ID must be set if AMI is set")
 		}
 	}
+	if ch.Tenancy != "" {
+		catcher.NewWhen(ch.Tenancy != "default" && ch.Tenancy != "dedicated" && ch.Tenancy != "host", "if set, tenancy must be 'default', 'shared', or dedicated'")
```
Comment (bynn) [inline apimodels/agent_models.go:250]: nit: would it be worth making consts for these 3 things? maybe not haha

File: docs/Project-Configuration/Project-Commands.md [hunk 2]
```diff
@@ -606,6 +606,8 @@ EC2 Parameters:
     set. May set if `distro` is set, which will override the value from
     the distro configuration.
 -   `subnet_id` - Subnet ID for the VPC. Must be set if `ami` is set.
+-   `tenancy` - If set, defines how the hosts are distributed across
```
Comment (bynn) [inline docs/Project-Configuration/Project-Commands.md:609]: might be obvious but can we say that it will be set to default if not specified?

**Review by bynn — APPROVED** [review]
2 unimportant nits

File: apimodels/agent_models.go (hunk 1 above)
Comment (Kimchelly) [reply apimodels/agent_models.go]: Hmm yeah it's a good idea. I added constants.

File: docs/Project-Configuration/Project-Commands.md (hunk 2 above)
Comment (Kimchelly) [reply docs/Project-Configuration/Project-Commands.md:609]: Sure, I mentioned the default.

**Pull request merged**